KenticoOnboardingApplication/test/KenticoOnboardingApplication.Api.Tests/Controllers/ListControllerIntegrationTest.cs

```diff
@@ -0,0 +1,96 @@
+using System.Threading.Tasks;
+using System.Web.Http;
+using KenticoOnboardingApplication.Api.Controllers;
+using KenticoOnboardingApplication.Contracts.Helpers;
+using KenticoOnboardingApplication.Contracts.Models;
+using KenticoOnboardingApplication.Contracts.Repositories;
+using KenticoOnboardingApplication.Tests.Base;
+using KenticoOnboardingApplication.Tests.Base.Factories;
+using NSubstitute;
+using NUnit.Framework;
+using Unity;
+
+namespace KenticoOnboardingApplication.Api.Tests.Controllers
+{
+    [TestFixture]
+    public class ListControllerIntegrationTest
+    {
+        [Test]
+        public async Task PutItemAsync_ShouldCallRepositoryOnce()
+        {
+            var container = CreateContainer();
+            var controller = ResolveController(container);
+            var (repository, timeManager) = ResolveRepositoryAndTimeManager(container);
+            var (originItem, expectedItem, itemToPut) = CreateOriginExpectedAndItemToPut();
+            ArrangeRepositoryAndTimeManager(repository, timeManager, originItem, expectedItem);
+
+            var result = await controller.PutItemAsync(itemToPut.Id, itemToPut);
+            var executedResult = await result.ExecuteAsync(CancellationToken.None);
+            executedResult.TryGetContentValue(out Item resultItem);
+
+            await repository.Received(1).GetItemAsync(originItem.Id);
+            await repository.DidNotReceive().GetItemAsync(Arg.Is<Guid>(id => id != originItem.Id));
+            await repository.Received(1).UpdateItemAsync(Arg.Is<Item>(item => item.Id == originItem.Id));
+            await repository.DidNotReceive().UpdateItemAsync(Arg.Is<Item>(item => item.Id != originItem.Id));
+            await repository.DidNotReceive().GetAllItemsAsync();
+            await repository.DidNotReceive().DeleteItemAsync(Arg.Any<Guid>());
+            await repository.DidNotReceive().AddItemAsync(Arg.Any<Item>());
+            timeManager.Received(1).GetCurrentTime();
+            Assert.That(resultItem, Is.EqualTo(expectedItem).UsingItemComparer());
+            Assert.That(executedResult.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+        }
+
+        private static void ArrangeRepositoryAndTimeManager(IListRepository repository, ITimeManager timeManager,
+            Item originItem, Item expectedItem)
+        {
+            repository.UpdateItemAsync(Arg.Is<Item>(item => item.Id == originItem.Id))
+                .Returns(result => Task.FromResult((Item) result[0]));
+            repository.GetItemAsync(originItem.Id).Returns(Task.FromResult(originItem));
+            timeManager.GetCurrentTime().Returns(expectedItem.LastUpdateTime);
+        }
+
+        private static (Item originItem, Item expectedItem, Item itemToPut) CreateOriginExpectedAndItemToPut()
+        {
+            const string lastUpdateTime = "2018-12-24";
+            const string id = "00000000-0000-0000-0000-000000000005";
+            var originItem = ItemsCreator.CreateItem(id: id, text: "text");
```

Comment:
```suggestion
            var originalItem = ItemsCreator.CreateItem(id: id, text: "text");
```
I believe it would either have to be `itemOfOrigin` or `originalItem`. If you were trying to find something short, you might use `primeItem` if you wish (or other [synonym](https://www.thesaurus.com/browse/original?s=t) you find fitting 😺)